homeassistant/components/stream/recorder.py

```diff
@@ -50,7 +51,11 @@ def recorder_save_worker(file_out: str, segments: deque[Segment]):
         last_sequence = segment.sequence
 
         # Open segment
-        source = av.open(segment.segment, "r", format=SEGMENT_CONTAINER_FORMAT)
+        source = av.open(
```

Comment:
Since the data is now stored in a bunch of `bytes` objects but `pyav` works on file objects, we reconstruct a `BytesIO` object from the `bytes` objects.